resources/bucket_list.py

```diff
@@ -0,0 +1,245 @@
+                      'uri':fields.Url('bucket_list.bucket'),
+                    }
+
+bucket_list_blueprint = Blueprint('bucket_list', __name__)
+api_bucket_list = Api(bucket_list_blueprint)
+
+@auth.verify_token
+def verify_token(token):
+    """"token authentication"""
+    # authenticate by token
+    user = Users.verify_auth_token(token)
+    if not user:
+        return False
+    g.user = user
+    return True
+
+class BucketListAPI(Resource):
+    """ bucket list API """
+
+    decorators = [auth.login_required]
+
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('name', type=str, required=True,
+                                   help='No buckelist name provided',
+                                   location='json')
+        super(BucketListAPI, self).__init__()
+
+    def get(self, id=None):
+        """ fxn for get request of bucketlist or bucketlists """
+
+        if id:
+            user_id = g.user.id
+            bucket_list = BucketList.query.filter_by(id=id, created_by=user_id).all()
+            if bucket_list:
+                return marshal(bucket_list, bucket_list_field), 200
+
+            return {'error':'bucket list does not exist'}, 404
+        else:
+            self.reqparse = reqparse.RequestParser()
+            self.reqparse.add_argument('page', location="args", type=int, required=False, default=1)
+            self.reqparse.add_argument('per_page', location="args", type=int, default=20)
+            self.reqparse.add_argument('q', location="args", required=False)
+
+            args = self.reqparse.parse_args()
+            search = args['q']
+            page = args['page']
+            per_page = args['per_page']
+            user_id = g.user.id
+
+            if search:
+                bucket_list = BucketList.query.filter(BucketList.created_by == user_id,\
+                                BucketList.name.like('%' + search + '%')).\
+                                paginate(page, per_page, False)
+            else:
+                bucket_list = BucketList.query.filter_by(created_by=user_id).\
+                              paginate(page, per_page, False)
+
+            if bucket_list:
+                if bucket_list.has_prev:
+                    url_next = request.url + '?page='+ str(page + 1) + '&limit=' + str(per_page)
+                else:
+                    url_next = 'Null'
+
+                if bucket_list.has_next:
+                    url_prev = request.url + '?page='+ str(page - 1) + '&limit=' + str(per_page)
+                else:
+                    url_prev = 'Null'
+
+                return {'meta':{'next_page':url_next,
+                                'previous_page':url_prev,
+                                'total_pages': bucket_list.pages},
+                        'bucketlist':marshal(bucket_list.items, bucket_list_field)}, 200
+
+            else:
+                return {'message': 'no bucket_list available'}, 200
+
```

Comment:
Instead you having nested `if else` use `if not bucket_list: return`. This will minimize nested if else statements